policy-controller/src/admission.rs

```diff
@@ -227,13 +236,20 @@ impl Validate<AuthorizationPolicySpec> for Admission {
             bail!("only a single NetworkAuthentication may be set");
         }
 
-        if mtls_authns_count + net_authns_count < spec.required_authentication_refs.len() {
+        if mtls_authns_count + sa_authns_count > 1 {
+            bail!("only MeshTLSAuthentication or ServiceAccount may be set");
+        }
+
```

Comment:
We can do this check before we do any of the net checks (i.e. move this up to 229).